samples/drivers/spi_nand/boards/stm32l562e_dk.overlay

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (c) 2022 Macronix International Co., Ltd.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+&spi3 {
+	pinctrl-0 = <&spi3_sck_pg9 &spi3_mosi_pb5 &spi3_miso_pb4>;
+	cs-gpios = <&gpioe 0 (GPIO_ACTIVE_LOW | GPIO_PULL_UP)>;
+	status = "okay";
+
+	mx31lf1ge4bc: spi_nand_flash@0 {
+		compatible = "jedec,spi-nand";
+		reg = <0>;
+		spi-max-frequency = <500000>;
+		label = "MX31LF1GE4BC";
+		size = <4194304>;
+		block-size = <2048>;
+		page-size = <2048>;
```

Comment:
> Not an "official" reviewer, but I used this PR as a basis to create a SPI driver for the Micron MT29F8G01. I found several issues in this implementation and will make some comments based on those findings.
> 

If you have read the doc for MT29F8G01, by the rules of internet discussions, you are at least "an expert".

Reply:
@bbooher Would it be possible to share your Micron driver? I'm in the same position and just started creating the Micron MT29F8G01 driver. You wrote that you found several issues with this driver. Have you mentioned all of them? I would appreciate any help.

Reply:
@coran21 - I threw this together rather quickly. https://github.com/bbooher/mt29f-zephyr-spi
The implementation is limited - I only use it for raw data logging (no file system support, etc.). The driver only supports page-aligned reads/writes, so your application layer needs to manage this. I guarantee no claims of efficacy or correctness, lol. I haven't looked at it in a while - it may be left in a mixed state of coding style, etc.

Reply:
@bbooher Great, thank you! I'll have to modify it to work with STM32 Octo SPI, but it's a great start.